packages/pluggableWidgets/rich-text-web/src/components/RichText.tsx

```diff
@@ -0,0 +1,112 @@
+import { createElement, ReactElement, useState, useEffect, useMemo } from "react";
+import { CKEditorHookProps, CKEditorType, CKEditorConfig, CKEditorEventAction } from "ckeditor4-react";
+import { getDimensions, Dimensions } from "@mendix/piw-utils-internal";
+import { defineEnterMode, addPlugin, PluginName } from "../utils/ckeditorConfigs";
+import sanitizeHtml from "sanitize-html";
+import classNames from "classnames";
+import { ReadOnlyStyleEnum, EnterModeEnum, ShiftEnterModeEnum, AdvancedConfigType } from "../../typings/RichTextProps";
+import { MainEditor } from "./MainEditor";
+
+export interface RichTextProps {
+    name: string;
+    readOnly: boolean;
+    spellChecker: boolean;
+    sanitizeContent?: boolean;
+    value: string | undefined;
+    advancedConfig: AdvancedConfigType[] | null;
+    plugins?: string[];
+    readOnlyStyle: ReadOnlyStyleEnum;
+    toolbar: CKEditorConfig;
+    enterMode?: EnterModeEnum;
+    shiftEnterMode?: ShiftEnterModeEnum;
+    editorType: CKEditorType;
+    dimensions?: Dimensions;
+    advancedContentFilter?: { allowedContent: string; disallowedContent: string } | null;
+    onValueChange?: (value: string) => void;
+    onKeyChange?: () => void;
+    onKeyPress?: () => void;
+}
+
+export const RichTextEditor = (props: RichTextProps): ReactElement => {
+    const {
+        editorType,
+        plugins,
+        enterMode,
+        shiftEnterMode,
+        value,
+        readOnly,
+        readOnlyStyle,
+        advancedContentFilter
+    } = props;
+    const [element, setElement] = useState<HTMLElement | null>(null);
+    const { width, height } = props.dimensions
+        ? getDimensions({ ...props.dimensions })
+        : {
+              width: "100%",
+              height: "100%"
+          };
+    const [ckeditorConfig, setCkeditorConfig] = useState<CKEditorHookProps<"change">>({
```

Comment:
This `CKEditorHookProps<"change">` seems a bit strange, sounds like an on change handler or something 🤔

Reply:
it's not controlled by me :) 
for more details check [CKEditor4 docs and react typings](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_react_v2.html#api-2)